lib/ModuleDependencyError.js

```diff
@@ -0,0 +1,20 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+*/
+var formatLocation = require("./formatLocation");
+
+function ModuleDependencyError(module, err, loc) {
+	Error.call(this);
+	Error.captureStackTrace(this, ModuleDependencyError);
+	this.name = "ModuleDependencyError";
+	this.message = formatLocation(loc) + " ";
+	this.details = err.stack.split("\n").slice(1).join("\n");
+	this.message += err.message;
+	this.origin = this.module = module;
+	this.error = err;
+}
```

Comment:
```js
class ModuleDependencyError extends Error {
  constructor(module, err, loc) {
    super(`${formatLocation(loc)} ${err.message}`);
    this.name = "ModuleDependencyError";
    this.details = err.stack.split("\n").slice(1).join("\n"); // err.stack.replace(/^./g, "")?
    this.origin = this.module = module;
    this.error = err;
  }
}
```

Calling `Error.captureStackTrace` is not necessary on native subclasses. The message can also just be given as the `super` argument instead of assigning to `this.message`. However, not calling `Error.captureStackTrace` does leave the constructor itself as the top of the stack trace.